js/algoliasearch/instantsearch.js

```diff
@@ -159,27 +159,44 @@ document.addEventListener("DOMContentLoaded", function (event) {
 		 * This widget renders all products into result page
 		 * Docs: https://community.algolia.com/instantsearch.js/documentation/#hits
 		 **/
-		search.addWidget(
-			algoliaBundle.instantsearch.widgets.hits({
-				container: '#instant-search-results-container',
-				templates: {
-					allItems: $('#instant-hit-template').html()
-				},
-				transformData: {
-					allItems: function (results) {
-						for (var i = 0; i < results.hits.length; i++) {
-							results.hits[i] = transformHit(results.hits[i], algoliaConfig.priceKey);
-							results.hits[i].isAddToCartEnabled = algoliaConfig.instant.isAddToCartEnabled;
-							
-							results.hits[i].algoliaConfig = window.algoliaConfig;
-						}
-						
-						return results;
+		if (algoliaConfig.instant.infiniteScroll.enabled === true) {
+			search.addWidget(
+				algoliaBundle.instantsearch.widgets.infiniteHits({
+					container: '#instant-search-results-container',
+					templates: {
+						empty: algoliaConfig.translations.noResults,
+						item: $('#instant-hit-template').html()
+					},
+					hitsPerPage: algoliaConfig.hitsPerPage,
+					showMoreLabel: algoliaConfig.translations.showMore,
+					cssClasses : {
+						root: 'clearfix'
 					}
-				},
-				hitsPerPage: algoliaConfig.hitsPerPage
-			})
-		);
+				})
+			);
+		} else {
+			search.addWidget(
+				algoliaBundle.instantsearch.widgets.hits({
+					container: '#instant-search-results-container',
+					templates: {
+						allItems: '{{#hits}}' + $('#instant-hit-template').html() + '{{/hits}}'
```

Comment:
That we cannot do, people need to have option to update template without touching JS code.
And it's a huge BC break as current users may have custom template and this code would make it not work at all.